test/ui/shared/test-url-util.js

```diff
@@ -0,0 +1,61 @@
+/* global describe, it */
+
+import expect from 'expect';
+import UrlUtil from '../../../ui/shared/url-util';
+
+describe('UrlUtil', function () {
+  describe('getHostname', function() {
+    it('returns hostname for full url', function() {
+      expect(UrlUtil.getHostname('http://example.com')).toEqual('example.com');
+    });
+    it('returns null for empty url', function() {
+      expect(UrlUtil.getHostname('')).toEqual(null);
+    });
+  });
+
+  describe('getScheme', function() {
+    it('is null for empty', function() {
+      expect(UrlUtil.getScheme('/file/path/to/file')).toEqual(null);
+    });
+
+    it('is null for localhost', function() {
+      expect(UrlUtil.getScheme('localhost://127.0.0.1')).toEqual(null);
+    });
+
+    it('gets scheme with :', function() {
+      expect(UrlUtil.getScheme('data:text/html,')).toEqual('data:');
+    });
+
+    it('gets scheme with ://', function() {
+      expect(UrlUtil.getScheme('http://www.mozilla.org')).toEqual('http://');
+    });
+  });
+
+  describe('prependScheme', function() {
+    it('prepends file:// to absolute file path', function() {
+      expect(UrlUtil.prependScheme('/file/path/to/file')).toEqual('file:///file/path/to/file');
+    });
+
+    it('defaults to http://', function() {
+      expect(UrlUtil.prependScheme('www.mozilla.org')).toEqual('http://www.mozilla.org');
+    });
+
+    it('keeps schema if already exists', function() {
+      expect(UrlUtil.prependScheme('https://www.mozilla.org')).toEqual('https://www.mozilla.org');
+    });
+  });
+
+  describe('isURL', function () {
```

Comment:
Should test the falsy scenario here, and aren't these technically URI's, not URLs?